Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm

```diff
@@ -142,7 +143,13 @@
     m_lastSentSizeToContentAutoSizeMaximumSize = m_webPageProxy.sizeToContentAutoSizeMaximumSize();
     m_lastSentSize = m_size;
     m_isWaitingForDidUpdateGeometry = true;
-    m_webPageProxy.sendWithAsyncReply(Messages::DrawingArea::UpdateGeometry(m_size, false /* flushSynchronously */, MachSendRight()), [weakThis = WeakPtr { this }] {
+
+    std::optional<VisibleContentRectUpdateInfo> visibleRectUpdateInfo;
+#if PLATFORM(IOS_FAMILY)
+    visibleRectUpdateInfo = m_webPageProxy.pageClient().createVisibleContentRectUpdateInfo();
```

Comment:
Do we send an `UpdateGeometry` message when the window size does not change?

I am wondering if we should only be sending this `visibleRectUpdateInfo` when the window size changes to avoid a potential performance regression on iOS? @hortont424 do you know?

Or will we always early return if duplicate `visibleRectUpdateInfo` are sent? Its not immediately clear to me looking at the implementation of `WebPage::updateVisibleContentRects`

Reply:
Seems like there's an early return in DrawingAreaProxy::setSize
(which is upstream of sendUpdateGeometry), so I think it's fine.